solver/src/main/java/org/chocosolver/solver/search/strategy/BlackBoxConfigurator.java

```diff
@@ -0,0 +1,403 @@
+
+    /**
+     * Strategy to use on integer variables (default is {@link Search#intVarSearch(IntVar...)}
+     */
+    Function<IntVar[], AbstractStrategy<IntVar>> intVarStrategy;
+
+    /**
+     * Strategy to use on integer objective.
+     */
+    BiFunction<IntVar, Boolean, AbstractStrategy<IntVar>> intObjVarStrategy;
+
+    /**
+     * Strategy to use on set variables (default is {@link Search#setVarSearch(SetVar...)}
+     */
+    Function<SetVar[], AbstractStrategy<SetVar>> setVarStrategy;
+
+    /**
+     * Strategy to use on graph variables (default is {@link Search#graphVarSearch(GraphVar...)}
+     */
+    Function<GraphVar<?>[], AbstractStrategy<GraphVar<?>>> graphVarSearch;
+
+    /**
+     * Strategy to use on real variables (default is {@link Search#realVarSearch(RealVar...)}
+     */
+    Function<RealVar[], AbstractStrategy<RealVar>> realVarStrategy;
+
+    /**
+     * Strategy to use on real objective.
+     */
+    BiFunction<RealVar, Boolean, AbstractStrategy<RealVar>> realObjVarStrategy;
+    /**
+     * Meta strategy to use (default is {@link Search#lastConflict(AbstractStrategy)}
+     */
+    Function<AbstractStrategy<?>, AbstractStrategy<?>> metaStrategy;
+
+    /**
+     * Restart strategy (default is "no restart")
+     */
+    Function<Solver, AbstractRestart> restartPolicy;
+
+    /**
+     * Set nogood recording in restart (default is true)
+     */
+    boolean nogoodOnRestart;
+
+    /**
+     * Add Generating Partial Assignment procedure (default is false)
+     */
+    boolean generatePartialAssignment;
+
+    /**
+     * Force restart on solution
+     */
+    boolean restartOnSolution;
+
+    /**
+     * Set to <i>true</i> to exclude views in the search strategy scope (default is <i>false</i>)
+     */
+    boolean excludeViews;
+
+    /**
+     * Exclude the objective variable, if any, from the search strategy.
+     */
+    boolean excludeObjective;
+
+
+    private BlackBoxConfigurator() {
+    }
+
+    /**
+     * Initialize the black box builder with default settings.
+     *
+     * @return return a pre-configured black box configurator
+     */
+    public static BlackBoxConfigurator init() {
+        return new BlackBoxConfigurator()
+                .setIntVarStrategy(Search::intVarSearch)
+                .setIntObjVarStrategy((obj, max) -> max ? Search.minDomUBSearch(obj) : Search.minDomLBSearch(obj))
+                .setSetVarStrategy(Search::setVarSearch)
+                .setGraphVarSearch(Search::graphVarSearch)
+                .setRealVarStrategy(Search::realVarSearch)
+                .setRealObjVarStrategy((obj, max) -> Search.realVarSearch(new Cyclic<>(), max ? new RealDomainMax() : new RealDomainMin(),
+                        !max, obj))
+                .setMetaStrategy(Search::lastConflict)
+                .setRestartPolicy(solver -> AbstractRestart.NO_RESTART)
+                .setRestartOnSolution(false)
+                .setNogoodOnRestart(false)
```

Comment:
Comments of the class indicate that it is valued to true by default. Either the comment or this value should me modified. And if set to true here, methods forCsp() and forCop() should be modified in order not to declare true a variable already set to true.